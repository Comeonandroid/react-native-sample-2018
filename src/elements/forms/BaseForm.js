import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { Formik } from 'formik'

// Wrapper over Formik. For more info check the docs https://github.com/jaredpalmer/formik

export default class BaseForm extends PureComponent {
  render() {
    const {
      enableReinitialize,
      initialValues,
      validationSchema,
      validate,
      onSubmit,
      render
    } = this.props
    return (
      <Formik
        enableReinitialize={enableReinitialize}
        initialValues={initialValues}
        validationSchema={validationSchema}
        validate={validate}
        onSubmit={onSubmit}
        render={render}
      />
    )
  }
}

BaseForm.propTypes = {
  render: PropTypes.func.isRequired,
  enableReinitialize: PropTypes.bool,
  initialValues: PropTypes.object,
  validationSchema: PropTypes.func,
  onSubmit: PropTypes.func,
  validate: PropTypes.func
}
