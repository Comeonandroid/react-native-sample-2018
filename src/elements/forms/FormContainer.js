import React from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'

import { View } from '../index'

const FormContainer = ({ children, ...rest }) => (
  <Element {...rest}>
    {children}
  </Element>
)

FormContainer.propTypes = {
  children: PropTypes.node
}

const Element = styled(View) `
  padding: 0px 12px;
`

export default FormContainer