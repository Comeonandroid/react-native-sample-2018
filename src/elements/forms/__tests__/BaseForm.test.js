import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import BaseForm from '../BaseForm'
import FormContainer from '../FormContainer'

const setup = propOverrides => {
  const props = Object.assign({
    initialValues: {},
    validationSchema: jest.fn(),
    validate: jest.fn(),
    render: () => (<FormContainer></FormContainer>),
    onSubmit: jest.fn()
  }, propOverrides)

  const component = <BaseForm {...props} />
  const wrapper = shallow(component)

  return {
    props,
    wrapper,
    component
  }
}

describe('<BaseForm />', () => {
  const { props, component, wrapper } = setup()
  itBehavesLike('aThemedComponent', { subject: component })

  it('pass props correctly', () => {
    expect(wrapper.prop('initialValues')).toEqual(props.initialValues)
    expect(wrapper.prop('validationSchema')).toEqual(props.validationSchema)
    expect(wrapper.prop('validate')).toEqual(props.validate)
    expect(wrapper.prop('render')).toEqual(props.render)
    expect(wrapper.prop('onSubmit')).toEqual(props.onSubmit)
  })
})
