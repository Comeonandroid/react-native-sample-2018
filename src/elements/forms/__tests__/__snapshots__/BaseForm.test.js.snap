// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseForm /> dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "paddingBottom": 0,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 0,
        },
        undefined,
      ],
    ]
  }
/>
`;

exports[`<BaseForm /> light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "paddingBottom": 0,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 0,
        },
        undefined,
      ],
    ]
  }
/>
`;
